node/src/components/linear_chain_sync/operations.rs

```diff
@@ -588,3 +612,179 @@ where
 
     Ok(trusted_header_output)
 }
+
+/// Returns `true` if `most_recent_block` belongs to an era that is still ongoing.
+fn is_current_era<I>(
+    most_recent_block: &BlockHeader,
+    maybe_switch_block: Option<&BlockHeader>,
+    chainspec: &Chainspec,
+    now: Timestamp,
+) -> Result<bool, LinearChainSyncError<I>>
+where
+    I: Eq + Debug,
+{
+    // Compute the start timestamp of the current era, and the number of blocks so far.
+    let (blocks_in_this_era, era_start) = match maybe_switch_block {
+        Some(switch_block) => (
+            most_recent_block.height() - switch_block.height(),
+            switch_block.timestamp(),
+        ),
+        None => (
+            most_recent_block.height() + 1,
+            chainspec
+                .protocol_config
+                .activation_point
+                .genesis_timestamp()
+                .ok_or_else(|| {
+                    error!(
+                        ?most_recent_block,
+                        "no recent switch block and no genesis timestamp"
+                    );
+                    LinearChainSyncError::MissingGenesisTimestamp
+                })?,
+        ),
+    };
+
+    // If the minimum era duration has not yet run out, the era is still current.
+    if now.saturating_diff(era_start) < chainspec.core_config.era_duration {
+        return Ok(true);
+    }
+
+    // Otherwise estimate the earliest possible end of this era based on how many blocks remain.
+    let remaining_blocks_in_this_era = chainspec
+        .core_config
+        .minimum_era_height
+        .saturating_sub(blocks_in_this_era);
+    let min_round_length = TimeDiff::from(1 << chainspec.highway_config.minimum_round_exponent);
+    let time_since_most_recent_block = now.saturating_diff(most_recent_block.timestamp());
+    Ok(time_since_most_recent_block < min_round_length * remaining_blocks_in_this_era)
```

Comment:
Do I understand correctly, that this will return `true` if we had just downloaded the latest block and there will be no more rounds in the era?

Reply:
I don't think so: `remaining_blocks_in_this_era` would be `0`, so this would be `false`?
The third and sixth scenario in `test_is_current_era` should confirm that.

Reply:
Of course :facepalm: :see_no_evil: :bat: